src/wallet/test/wallet_tests.cpp

```diff
@@ -69,32 +68,30 @@ BOOST_AUTO_TEST_CASE(coin_selection_tests)
     CoinSet setCoinsRet, setCoinsRet2;
     CAmount nValueRet;
 
-    LOCK(wallet.cs_wallet);
-
     // test multiple times to allow for differences in the shuffle order
     for (int i = 0; i < RUN_TESTS; i++)
     {
         empty_wallet();
 
         // with an empty wallet we can't even pay one cent
-        BOOST_CHECK(!wallet.SelectCoinsMinConf( 1 * CENT, 1, 6, vCoins, setCoinsRet, nValueRet));
+        BOOST_CHECK(!CWallet::SelectCoinsMinConf( 1 * CENT, 1, 6, vCoins, setCoinsRet, nValueRet));
```

Comment:
This looks _very_ suspicious to me. Does this work? Does this method actually not use any properties of `CWallet`?

Reply:
Yes. That was the whole thing that I discovered - that this method should have always been static.